SETUP.md

```diff
@@ -0,0 +1,101 @@
+  - Open command prompt (Open Start Menu then type <code>cmd</code> and press [Enter])</li>
+  - Type the following line into the command prompt window exactly as shown:
+    `setx HOME "%USERPROFILE%"`
+  - Press [Enter], you should see `SUCCESS: Specified value was saved.`
+  - Quit command prompt by typing `exit` then pressing [Enter]
+
+This will provide you with Bash in the Git Bash program.
+
+### macOS
+
+The default shell in all versions of macOS is Bash, so no need to install 
+anything. You access Bash from the Terminal (found in `/Applications/Utilities`).
+See the Git installation <a href="https://www.youtube.com/watch?v=9LQhwETCdwY ">
+video tutorial</a> for an example on how to open the Terminal. You may want to 
+keep Terminal in your dock for this workshop.
+
+### Linux
+
+The default shell is usually Bash, but if your machine is set up differently you 
+can run it by opening a terminal and typing `bash`. There is no need to install 
+anything.
+
+## Text Editor
+
+When you're writing code, it's nice to have a text editor that is optimized for 
+writing code, with features like automatic color-coding of key words.  The 
+default text editor on macOS and Linux is usually set to Vim, which is not 
+famous for being intuitive. if you accidentally find yourself stuck in it, try 
+typing the escape key, followed by `:q!` (colon, lower-case 'q', exclamation 
+mark), then hitting Return to return to the shell.
+
+### Windows
+
+<a href="https://www.youtube.com/watch?v=339AEqk9c-8">Video Tutorial</a>
+
+nano is a basic editor and the default that instructors use in the workshop.
+To install it, download the <a href="https://github.com/swcarpentry/windows-installer/releases/download/v0.3/SWCarpentryInstaller.exe">
+Software Carpentry Windows installer</a> and double click on the file to run it.
+This installer requires an active internet connection.
+
+Others editors that you can use are:
+
+- <a href="http://notepad-plus-plus.org/">Notepad++</a> or
+- <a href="http://www.sublimetext.com/">Sublime Text</a>.
+
+**Be aware that you must add its installation directory to your system path.**
+Please ask your instructor to help you do this.
+
+### macOS
+
+nano is a basic editor and the default that instructors use in the workshop.
+See the Software Carpentry Git installation 
+<a href="https://www.youtube.com/watch?v=9LQhwETCdwY ">video tutorial</a>
+for an example on how to open nano. It should be pre-installed.
+
+Others editors that you can use are:
+
+- <a href="http://www.barebones.com/products/textwrangler/">Text Wrangler</a> or
+- <a href="http://www.sublimetext.com/">Sublime Text</a>.
+
+### Linux
+
+nano is a basic editor and the default that instructors use in the workshop. It 
+should be pre-installed.
+
+Others editors that you can use are:
+
+- <a href="https://wiki.gnome.org/Apps/Gedit">Gedit</a>,
+- <a href="http://kate-editor.org/">Kate</a> or
+- <a href="http://www.sublimetext.com/">Sublime Text</a>.
```

Comment:
Perhaps we can recommend everyone to get sublime text as an optional text editor just for copy pasting code and IP addresses around? I don't think it is critical.

Reply:
That is what this section is for....